tests/distributions/test_multivariate.py

```diff
@@ -13,6 +13,123 @@
+    def test_init_check_shape(self):
+        with self.test_session(use_gpu=True):
+            with self.assertRaisesRegexp(ValueError, "should have rank"):
+                MultivariateNormalCholesky(tf.zeros([]), tf.zeros([]))
+            with self.assertRaisesRegexp(ValueError, "should have rank"):
+                MultivariateNormalCholesky(tf.zeros([1]), tf.zeros([1]))
+            with self.assertRaisesRegexp(ValueError, 'compatible'):
+                MultivariateNormalCholesky(
+                    tf.zeros([1, 2]), tf.placeholder(tf.float32, [1, 2, 3]))
+            u = tf.placeholder(tf.float32, [None])
+            len_u = tf.shape(u)[0]
+            dst = MultivariateNormalCholesky(
+                tf.zeros([2]), tf.zeros([len_u, len_u]))
+            with self.assertRaisesRegexp(
+                    tf.errors.InvalidArgumentError, 'compatible'):
+                dst.sample().eval(feed_dict={u: np.ones((3,))})
+
+    def test_shape_inference(self):
+        with self.test_session(use_gpu=True):
+            # Static
+            mean = 10 * np.random.normal(size=(10, 11, 2)).astype('d')
+            cov = np.zeros((10, 11, 2, 2))
+            dst = MultivariateNormalCholesky(
+                tf.constant(mean), tf.constant(cov))
+            self.assertEqual(dst.get_batch_shape().as_list(), [10, 11])
+            self.assertEqual(dst.get_value_shape().as_list(), [2])
+            # Dynamic
+            unk_mean = tf.placeholder(tf.float32, None)
+            unk_cov = tf.placeholder(tf.float32, None)
+            dst = MultivariateNormalCholesky(unk_mean, unk_cov)
+            self.assertEqual(dst.get_value_shape().as_list(), [None])
+            feed_dict = {unk_mean: np.ones(2), unk_cov: np.eye(2)}
+            self.assertEqual(list(dst.batch_shape.eval(feed_dict)), [])
+            self.assertEqual(list(dst.value_shape.eval(feed_dict)), [2])
+
+    def _gen_test_params(self, seed):
+        np.random.seed(seed)
+        mean = 10 * np.random.normal(size=(10, 11, 3)).astype('d')
+        cov = np.zeros((10, 11, 3, 3))
+        cov_chol = np.zeros_like(cov)
+        for i in range(10):
+            for j in range(11):
+                cov[i, j] = stats.invwishart.rvs(3, np.eye(3))
+                cov[i, j] /= np.max(np.diag(cov[i, j]))
+                cov_chol[i, j, :, :] = np.linalg.cholesky(cov[i, j])
+        return mean, cov, cov_chol
+
+    def test_sample(self):
+        tf.set_random_seed(233)
+        with self.test_session(use_gpu=True):
+            def test_sample_with(seed):
+                mean, cov, cov_chol = self._gen_test_params(seed)
+                dst = MultivariateNormalCholesky(
+                    tf.constant(mean), tf.constant(cov_chol))
+                n_exp = 20000
+                samples = dst.sample(n_exp)
+                sample_shape = (n_exp, 10, 11, 3)
+                samples.shape.assert_is_compatible_with(
+                    tf.TensorShape(sample_shape))
```

Comment:
should be equal by `as_list` to make sure there is no `None` entries in the shape.